app/mm-redux/reducers/entities/groups.ts

```diff
@@ -194,7 +194,20 @@ function groups(state: Dictionary<Group> = {}, action: GenericAction) {
         }
         return nextState;
     }
+    case GroupTypes.RECEIVED_ALL_GROUPS_ASSOCIATED_TO_CHANNELS_IN_TEAM: {
+        const nextState = {...state};
+        const {groupsByChannelId} = action.data;
+
+        for (const group of Object.values(groupsByChannelId) as Group[]) {
+            if (group) {
+                nextState[group.id] = group;
+            }
+        }
+        return nextState;
```

Comment:
same comment as with the other reducer, only return the updated state if  something changed... also what about those that were removed...

Reply:
It doesnt really matter here if the groups in this list have extra groups, since this is the set of all possible groups on the server not just the team. The list of groups per channel is used to grab the groups from here so you would never end up displaying groups that dont need to be displayed.

As for updating the state _only_ if something changed is it worth it to compare all the groups received over here @enahum? since its an object you would have stringify two objects everytime and compare before reassigning, how does the performance of stringifying potentially hundreds of objects compare to a re-render?

Reply:
@fmunshi that is exactly what I told HM, review if doing the check is less or more expensive and keep it or change it based on the findings...

Remember that updating the state does trigger mapStateToProps of all mounted components + the selectors listening for this data changes

Reply:
Spoke with Elias. Came to the conclusion that we can leave this as is.